pandas/core/indexes/range.py

```diff
@@ -197,9 +197,6 @@ def _format_data(self, name=None):
         # we are formatting thru the attributes
         return None
 
-    def _format_with_header(self, header, na_rep="NaN") -> List[str]:
-        return header + [pprint_thing(x) for x in self._range]
-
```

Comment:
The added tests revealed that this made the output from ``.format`` different than for ``Int64Index``:

```python
>>> pd.RangeIndex(0, 18, 2).format()
['0', '2', '4', '6', '8', '10', '12', '14', '16']
>>> pd.Int64Index(range(0, 18, 2)).format()
 ['0 ', '2 ', '4 ', '6 ', '8 ', '10', '12', '14', '16']
```
 Notice the extra space for one-digit scalars in the ``Int64Index`` case. The output are identical after merging this PR.